definitions-trimmed-migrate/src/services/azure-storage-service/AzureStorageService.ts

```diff
@@ -0,0 +1,132 @@
+import { BlobServiceClient, ContainerClient } from '@azure/storage-blob';
+import { DequeuedMessageItem, QueueClient, QueueServiceClient } from '@azure/storage-queue';
+
+import { LogService } from '../log-service/LogService';
+import { RedisService } from '../redis-service/RedisService';
+import { AzureStorageOptions } from './types';
+
+export class AzureStorageService {
+    private blobServiceClient: BlobServiceClient;
+    private containerClient: ContainerClient;
+    private queueClient: QueueClient;
+
+    constructor(
+        private options: AzureStorageOptions,
+        private redisService: RedisService,
+        private logService: LogService
+    ) {
+        this.blobServiceClient = BlobServiceClient.fromConnectionString(
+            this.options.blobConnectionString
+        );
+
+        this.containerClient = this.blobServiceClient.getContainerClient(
+            this.options.blobContainerName
+        );
+
+        this.queueClient = QueueServiceClient.fromConnectionString(
+            this.options.queueConnectionString,
+            { retryOptions: { maxTries: 4 } }
+        ).getQueueClient(this.options.queueName);
+    }
+
+    public async iterateAndQueueAllBlobs() {
+        const startTime = new Date();
+        let i = 1;
+        let queued = 0;
+
+        try {
+            for await (const blob of this.containerClient.listBlobsFlat()) {
+                const { name, properties } = blob;
+                const { lastModified } = properties;
+
+                if (lastModified < startTime && !(await this.redisService.get(name))) {
+                    await this.queueMessage(name);
+                    await this.redisService.set(name, 1);
+                    queued++;
+                }
+
+                this.logService.log(
+                    'AzureStorageService.iterateAndQueueAllBlobs',
+                    `Iterated ${i++} and queued ${queued} blobs to be processed.`
+                );
+            }
+        } catch (err) {
+            this.logService.error('AzureStorageService.iterateAndQueueAllBlobs', err, {
+                exception: err as Error,
+            });
+
+            throw err;
+        }
+    }
+
+    public async receiveMessages(count: number) {
+        try {
+            return await this.queueClient.receiveMessages({ numberOfMessages: count });
```

Comment:
Is the default visibility timeout long enough for our download and subsequent processing?

Reply:
changed to 900 seconds (15 min) to be safe